src/main/java/net/dv8tion/jda/core/requests/IGatewayProvider.java

```diff
@@ -0,0 +1,21 @@
+package net.dv8tion.jda.core.requests;
+
+/**
+ * Tells JDA to which URL it should connect the websocket. Custom implementations can be used to have
```

Comment:
~~Tells~~ Provides

Reply:
`Provides JDA a strategy to determine a URL which it should use when connecting the Main Websocket`